DOC: documents about K8s deployment (#131)
xprobe-inc/xorbits: doc/source/deployment/cloud.rst

.. _deployment_cloud:

================
Cloud deployment
================

Prerequisites
-------------
Currently we support deploying Xorbits to Amazon EKS cluster.

Install Xorbits on the machine where you can control your Amazon EKS cluster.
Refer to :ref:`installation document <installation>`.

Amazon EKS
----------

> Firstly, make sure you have an EKS cluster and it can access `our Dockerhub <https://hub.docker.com/repository/docker/xprobe/xorbits>`_ to pull the Xorbits image.

Reviewer: ```suggestion
Firstly, make sure your EKS cluster can access `our Dockerhub <https://hub.docker.com/repository/docker/xprobe/xorbits>`_.
```

Secondly, install the `AWS Load Balancer Controller <https://docs.aws.amazon.com/eks/latest/userguide/aws-load-balancer-controller.html>`_.

Then, deploy Xorbits cluster, for example:

.. code-block:: python

    from kubernetes import config
    from xorbits.deploy.kubernetes import new_cluster
    cluster = new_cluster(config.new_client_from_config(), worker_cpu=1, worker_mem='4g', cluster_type='eks')

Note that the option ``cluster_type='eks'`` cannot be ignored.

> When a log of the form ``Xorbits endpoint http://<ingress_service_ip>:80 is ready!`` appears,

Reviewer: ```suggestion
You should be able to see ``Xorbits endpoint http://<ingress_service_ip>:80 is ready!`` soon, and
```

> you can access the web page of your xorbits cluster through the endpoint in the log.

Reviewer: ```suggestion
you can access the web UI of your Xorbits cluster using the endpoint.
```

.. _deployment_cloud:

================
Cloud deployment
================

Prerequisites
-------------
Currently, we support deploying Xorbits on an existing Amazon EKS cluster.

Install Xorbits on the machine for Amazon EKS cluster management.
Refer to :ref:`installation document <installation>`.

Amazon EKS
----------
Firstly, make sure your EKS cluster can access `our Dockerhub <https://hub.docker.com/repository/docker/xprobe/xorbits>`_ to pull the Xorbits image.

Secondly, install the `AWS Load Balancer Controller <https://docs.aws.amazon.com/eks/latest/userguide/aws-load-balancer-controller.html>`_.

Then, deploy Xorbits cluster, for example:

.. code-block:: python

    from kubernetes import config
    from xorbits.deploy.kubernetes import new_cluster
    cluster = new_cluster(config.new_client_from_config(), worker_cpu=1, worker_mem='4g')

> Note that the option ``cluster_type`` of the ``new_cluster`` api equals to ``auto`` by default, which means that

Reviewer: ```suggestion
Note that the option ``cluster_type`` of the function ``new_cluster`` has default value ``auto``, which means that
```

> the ``kubectl`` context on your machine will be detected. Please make sure kubectl uses your eks context.

Reviewer: ```suggestion
Xorbits will detect the ``kubectl`` context automatically. Please make sure ``kubectl`` is using the correct EKS context.
```
